### ISSUE TITLE
Use issues for team communication

### ISSUE THREAD

**Issue by dward2** [SylviaCHS/BME547_Final_Project#15]
@tiffanylin43 @SylviaCHS @kkl18 One of the things we will be looking for in this assignment is the use of comments in the GitHub issues as a way of intra-team communication.  So, as you are working over the next few days, use those comments as a way of communicating on some things.  Also, I see some of your milestones either don't have any issues, or only one or two assigned to them.  I'd add a few more.

**Comment by SylviaCHS** [comment]
@kkl18 Hi Kim, are you working on the server code? I plan to create a new branch. Could you push what you have now so I can create  a pull request to merge our server code? Thanks!

**Comment by KimberlyLennox** [comment]
Hi Sylvia,

I'm done with the code for today but the PEP8 checks will probably fail before merging. I can fix it first thing tomorrow and will update the issue then.

Regards,
Kim Lennox

On Apr 24, 2019 9:20 PM, SylviaCHS <notifications@github.com> wrote:

@kkl18<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_kkl18&d=DwMCaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=tZprIgxWmPtnvcVpMrhE0fQ1mDBan3FOHVvyVWW1ObA&m=cXohEvcblvsGLqqrXI86F_MaBSiYfxIsdrVHDK5_Dpw&s=Phz9G6OFYdhsDpvuH3t1fkVYLhxZwUUjJTEmH_HcG_I&e=> Hi Kim, are you working on the server code? I plan to create a new branch. Could you push what you have now so I can create a pull request to merge our server code? Thanks!

**Comment by KimberlyLennox** [comment]
Hi @SylviaCHS,

The checks are done, so the branch can be merged whenever you're ready. You should be able to check if processed images are showing up in the GUI, but some database modifications still need to be made.

**Comment by SylviaCHS** [comment]
@kkl18 Hi Kim, I am linking the client code to GUI for displaying processed image. Since one image upload will only be linked to 1 process method, can we add a key about process method in "/api/new_image" ? So that we don't need the "/api/process_image" route. 

Also, how should we handle the request of the same image upload with different process method?

**Comment by KimberlyLennox** [comment]
@sylviaCHS Hi Sylvia, how are you planning to separate the new image queries from the processed image queries? Remember that raw images need to be uploaded to the GUI as well.

________________________________

**Comment by SylviaCHS** [comment]
@kkl18 Could you upload your Mongo.py code? I think the one in the branch is from your last project. Thank you!

**Comment by SylviaCHS** [comment]
@kkl18 never mind! I was looking at the docstring and thought it is the last one.

**Comment by SylviaCHS** [comment]
Hello Dr. Ward @dward2, I am developing the GUI. It was working great but suddenly the mongoDB server responds really slowly. It used to take a few seconds, but now it would take ~minute for the same command of saving an image using the exact same code. Do you know what might cause the problem?

Thanks,
Sylvia

**Comment by dward2** [comment]
If it is working, but just slow, them I'm not sure.  Are your teammates also have slow access times to MongoDB?  It could just be that their servers are overloaded, or some are down, leading to increased connect time.  You said the change was sudden.  But, if it was gradual, perhaps there could be an issue with your database getting larger and requiring more time to manage.  But, if the change was sudden, this wouldn't be it.  But, you could always log into MongoDB and delete any old records to see if a small size would go faster.

**Comment by SylviaCHS** [comment]
@kkl18 @tiffanylin43
I converted the histogram and bins from array to list in Tiffany's code, it should work now. 
Also, the GUI code is working. It can take in a single image, display it, and save the processed image to a local folder. Feel free to try it out! We still need to make GUI to handle multiple files and display error.
I set up my own MongoDB and it is much faster. So I did change the connect address in 'mongo.py', feel free to change it after merging my brach.
